### DISCUSSION TITLE
Deploy

### DISCUSSION THREAD

**Discussion by waghanza** [SuspiciousLookingOwl/web-frameworks-benchmark#3] [Show and tell]
Hi @SuspiciousLookingOwl ,

How is this project deployed ? Just wonder if we could / should deploy in a automated way.

Regards,

**Comment by SuspiciousLookingOwl** [comment]
This project is deployed manually to Netlify. Setting up automated deployment in Netlify is possible and easy, however free account have a limitation of 300 minutes of build time per month.

**Reply by waghanza** [reply]
Yeah, indeed netlify have some limitation. But the idea behind this is : should we deploy automatically ? or not ?

PS : Their is plenty of services like netlify

**Reply by SuspiciousLookingOwl** [reply]
I don't see the reason why not😁

**Reply by waghanza** [reply]
I'm in favor of deploying from tags not **master**.

I mean this allow :
- to ensure quality of service
- to have a `sandbox` like (issue from prerelease tags from github)

**Reply by waghanza** [reply]
maybe https://surge.sh/ is a good idea. I can work on that

**Reply by waghanza** [reply]
any feedback on this service @SuspiciousLookingOwl ?

any idea @the-benchmarker/web-frameworks ?

**Reply by SuspiciousLookingOwl** [reply]
How about https://vercel.com? Seems to be more reputable than surge (I never heard of surge before 😄), the limitation for the free plan in vercel is 100 deployment / day (which should be plenty for us) with maximum build time / deploy is 45 minutes (which is also plenty for our app).

Also, a bit off topic, any plan on buying / renting custom domain for the website?

**Reply by System-Glitch** [reply]
It is purely static or does it need a back-end? Github pages may be enough if it's only static.

**Reply by SuspiciousLookingOwl** [reply]
It's purely static, but we need a provider that can run build script before deploying, and AFAIK GitHub pages doesn't support that

**Reply by System-Glitch** [reply]
It does. I'm using github pages in combination with github actions to publish vue.js / vuepress websites.

**Reply by SuspiciousLookingOwl** [reply]
I just checked and it does support build script!

**Reply by waghanza** [reply]
I prefer to use an other service than github. Mostly for 2 reasons :
- this project (at least for me) is about learning, it is 👌🏻 if I can learn
- I prefer to mix several features in the same service

btw, not opposed to use GH, the above arguments are mainly something like guidelines for me
